ansible/roles/status-report-install/tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+---
+- name: Create application folder
+  file:
+    name: "/opt/status/src"
+    state: directory
+    owner: "{{ ansible_user }}"
+- name: Retrieve application source files
+  git:
+    clone: yes
+    force: yes
+    dest: "/tmp/status"
+    repo: "http://github.com/jimrigsbee/api.git"
+- name: Copy application files
+  copy:
+    dest: "/opt/status/src/{{ item }}"
+    src: "/tmp/status/{{ item }}"
+    owner: "{{ ansible_user }}"
+    mode: 0644
+    remote_src: true
+  with_items:
+    - requirements.txt
+    - server.py
+    - wsgi.py
+- name: Upgrade pip
+  pip:
+    name: pip
+    state: latest
+- name: Install python virtualenv
+  pip:
+    name: virtualenv
+    state: latest
+- name: Allow nginx to access non-standard port
+  shell: |
+    semanage port -a -t http_port_t -p tcp 30904
+    semanage port -m -t http_port_t -p tcp 30904
+
+- name: Install python application as a service
+  include_role:
+    name: Stouts.wsgi
```

Comment:
How does Ansible find Stouts.wsgi here?

Reply:
It looks like this comes from the requirements.yml defined in the ocp4-workshop config. Does that mean that any config that wants to use the status-reporting role has to know to add stouts to it's requirements?

Reply:
Maybe we should do something more like this instead: https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-dependencies

Reply:
Hopefully you find this in ansible-skylight and ocp4-workshop.  For now this is the way it works in AgnosticD - you have to put your required roles for whatever you are doing in the config/*/requirements.yml